File: src/leetcode/solved-collection/876. Middle of the Linked List/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        int count = 0;
        ListNode temp = head;
        while(temp.next != null) {
            ++count;
            temp = temp.next;
        }

        int mid = (count % 2 == 0) ? (count / 2) : (count / 2) + 1;

        for(int i = 0; i < mid; i++) {
            head = head.next;
        }

        return head;
    }
}

File: src/leetcode/solved-collection/876. Middle of the Linked List/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        count = 0
        temp = head
        while temp.next:
            temp = temp.next
            count += 1

        mid = int(count / 2) if count % 2 == 0 else int(count / 2) + 1

        for i in range(mid):
            head = head.next

        return head
